examples/rich-text/index.js

```diff
@@ -70,7 +70,9 @@ class RichText extends React.Component {
 
   hasMark = (type) => {
     const { state } = this.state
-    return state.marks.some(mark => mark.type == type)
+    const characterMarks = state.marksPerCharacter
+    return characterMarks.length &&
+      characterMarks.every(charMarks => charMarks.some(mark => mark.type == type))
```

Comment:
I now see what you mean about why `state.marks` isn't enough by itself. But it does look like this kind of computation is going to get expensive, since people will always be using anonymous functions which are hard to memoize.

I wonder if we should instead introduce `state.hasMark(mark)` and `state.hasMarkByType(type)` methods that we can memoize ourselves?